utils/build-script-impl

```diff
@@ -1001,138 +998,245 @@ if [[ ! "${SKIP_BUILD_SWIFTPM}" ]] ; then
-            if [[ "${VALIDATION_TEST}" == "1" || "${LONG_TEST}" == "1" ]] ; then
-                SWIFT_STDLIB_TARGETS+=("swift-stdlib-${deployment_target}")
-            else
-                SWIFT_STDLIB_TARGETS+=("swift-test-stdlib-${deployment_target}")
-            fi
+done
+
+ALL_HOSTS=("${LOCAL_HOST}" "${CROSS_COMPILE_HOSTS[@]}")
+
+function is_cross_tools_host() {
+    local host="$1"
+    for t in "${CROSS_COMPILE_HOSTS[@]}" ; do
+        if [ "${host}" == "${t}" ] ; then
+            echo 1
         fi
+    done
+}
+
+# We only install in to host-specific subdirectories if there are cross-compile
+# hosts defined. Other users will expect their products at INSTALL_DESTDIR.
+function has_cross_compile_hosts() {
+    if [[ ${#ALL_HOSTS[@]} -gt 1 ]]; then
+        echo "1"
     fi
-    if [[ "${build_benchmark_this_target}" ]] ; then
-        SWIFT_BENCHMARK_TARGETS+=("swift-benchmark-${deployment_target}")
-        if [[ $(not ${SKIP_TEST_BENCHMARK}) ]] ; then
-          SWIFT_RUN_BENCHMARK_TARGETS+=("check-swift-benchmark-${deployment_target}")
-        fi
+}
+
+# When building cross-compilers for these hosts, merge all of their contents together with lipo
+function should_include_host_in_lipo() {
+    local host="$1"
+    if [[ -z "${SKIP_MERGE_LIPO_CROSS_COMPILE_TOOLS}" ]]; then
+        case ${host} in 
+            iphone* | appletv* | watchos* )
+                echo 1
+                ;;
+        esac
     fi
-    if [[ "${test_this_target}" ]] ; then
-        test_target_suffix=""
-        if [[ -n "${test_host_only}" ]] ; then
-            test_target_suffix="-non-executable"
-        fi
+}
+
+function host_has_darwin_symbols() {
+    local host="$1"
+    case ${host} in 
+        macosx* | iphone* | appletv* | watchos* )
+            echo 1
+            ;;
+    esac
+}
 
-        test_subset_target_suffix=""
-        if [[ "${VALIDATION_TEST}" == "1" ]] ; then
-            if [[ "${LONG_TEST}" == "1" ]] ; then
-                test_subset_target_suffix="-all"
+# FIXME: We currently do not support cross-compiling swift with compiler-rt.
+if [[ "${CROSS_COMPILE_HOSTS}" ]]; then
+    SKIP_COMPILER_RT=1
+fi
+
+STDLIB_DEPLOYMENT_TARGETS_DEFAULT="${STDLIB_DEPLOYMENT_TARGETS}"
+function get_stdlib_targets_for_host() {
+
+# FIXME: STDLIB_DEPLOYMENT_TARGETS argument assumed to apply when Host == Build
+# Cross-compile Hosts are only built with their native standard libraries.
+# To fix this, we would need to pass in a list of stdlib targets _per host_,
+# and the SWIFT_SDKS parameters would need to be able to capture both the SDK 
+# and architecture of each stdlib target -- currently it only captures the SDK.
+#
+# We turn these targets in to SWIFT_SDKS in `calculate_targets_for_host()`
+    
+    if [[ $(is_cross_tools_host $1) ]] ; then
+        echo "$1"
+    else
+        echo "${STDLIB_DEPLOYMENT_TARGETS_DEFAULT[@]}"
+    fi
+}
+
+function calculate_targets_for_host() {
+
+    SWIFT_STDLIB_TARGETS=()
+    SWIFT_SDKS=()
```

Comment:
How about instead of `swift-sdks` as a parameter to `build-script-impl`, we rename it to `build-stdlib-deployment-targets`. It would be more descriptive, and would specify the same kind of details as the existing `stdlib-deployment-targets` parameter. That would also allow us to only build specific slices (e.g. only `iphone-armv7`, instead of all `iphone-*` architectures), which I believe is something the core team is interested in.